src/dsl/verifier/proxy/stateHandler/setupStates.ts

```diff
@@ -1,14 +1,29 @@
 import logger from '../../../../common/logger';
-import { ProviderState, ProxyOptions } from '../types';
+import { ProxyOptions, ProviderState, ProviderStateV3 } from '../types';
+
+const isProviderStateV3 = (
+  descriptor: ProviderStateV3 | ProviderState
+): descriptor is ProviderStateV3 =>
+  (descriptor as ProviderStateV3).action !== undefined;
 
 // Lookup the handler based on the description, or get the default handler
 export const setupStates = (
-  descriptor: ProviderState,
+  descriptor: ProviderState | ProviderStateV3,
   config: ProxyOptions
 ): Promise<unknown> => {
   const promises: Array<Promise<unknown>> = [];
 
-  if (descriptor.states) {
+  if (isProviderStateV3(descriptor)) {
+    const handler = config.stateHandlers
```

Comment:
It might be better to do this check a bit more explicitly so that we can warn when no state handlers are present at all (and that check is probably relevant for both V3 and V2).

Reply:
Well, state handlers aren't mandatory - in that you can ignore them and the tests **might** still pass. 

In this way, it warns for each state. Are you suggesting we have a separate one that says "you don't have any state handlers registered, but there are states" type thing?

Reply:
Yeah, that’s what I was thinking. It feels like a “you probably are missing some config” situation.